automod/engine/context.go

```diff
@@ -54,6 +56,19 @@ type RecordOp struct {
 	Value any
 }
 
+// Originally intended for push notifications, but can also work for any inter-account notification.
+type NotificationContext struct {
+	AccountContext
+
+	Recipient    AccountMeta
+	Notification NotificationMeta
+}
+
+type NotificationMeta struct {
+	Reason  string
```

Comment:
You're on a roll with docs, don't drop it now! :D  What does this string typically look like, who provides it, and what sort of things should it affect?

(Or document that on the `NewNotificationContext` constructor and point at that from the struct comment.)